packages/std/src/results/response.rs

```diff
@@ -108,36 +101,69 @@ where
     }
 
     /// Add an attribute included in the main `wasm` event.
-    pub fn add_attribute<K: Into<String>, V: Into<String>>(&mut self, key: K, value: V) {
-        self.attributes.push(Attribute {
-            key: key.into(),
-            value: value.into(),
-        });
+    pub fn add_attribute(mut self, attr: impl Into<Attribute>) -> Self {
```

Comment:
I see why we need `.add_attribute(("foo", "bar"))` given Into.

I would love to see something like:

```rust
pub fn add_attribute<K, V>(mut self, key: K, value: V) -> Self 
 where (K, V): Into<Attribute> {
  // ...
}
```

but I guess there is no way with the Rust compiler to make such complex generics?

Reply:
It would just avoid a few extra parenthesis, but they do make the whole API feel a bit odd.

Since there is really only one `Into` defined (except from Self), why don't we simplify this:

```rust
impl<K: Into<String>, V: Into<String>> From<(K, V)> for Attribute { }
```

```rust
pub fn add_attribute(mut self, key: impl Into<String>, value: impl Into<String>) -> Self {
  self.attributes.push(Attribute::new(key, value));
  self
}
```

Other idea would be to have a second function that covers a common subset `add_attr<K, V>

Reply:
> but I guess there is no way with the Rust compiler to make such complex generics?

What you wrote there is 100% valid Rust as far as I'm aware! We can do that.

Reply:
Either of these two approaches look good to me.
They are the same currently, but the first one would possibly expand more in the future.

Neither support adding `Attribute` directly, but as I said above, I don't think that use case matters (and what matters is handled by the plural form)